ProxyStarcraft/ZergUnitExtensions.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Collections.Generic;
+
+namespace ProxyStarcraft
+{
+    public static class ZergUnitExtensions
+    {
+        public static void SpawnLarva(this ZergUnit queen, ZergBuilding hatchery, List<Command> commands)
+        {
+            if (queen == null || hatchery == null) // TODO: I had a queen die, I guess mid-update, causing a Null Ref.
+            {
+                return;
+            }
+            if (queen.Raw.Energy >= 25 && !hatchery.IsSpawningLarva() && hatchery.IsBuilt) // WELP: Spawn Larva puts a "build"-y action on Zerg town halls. The Zerg building is not "building" anything, but shows via Build Progress.
```

Comment:
Your comment confuses me.  I don't see that when I run this. Also also "IsBuilt" tells you about the build progress of the _Hatchery_, not what it's building. (Although presumably it is in fact a bad idea to try to spawn larvae on an unbuilt Hatchery.)

Reply:
The Zerg Bot logic builds queens based on Hatchery count, and will attempt to do just that. I think I was correct to put that check in, but for the wrong reason- when I wrote this comment, I don't think I knew about buffs. I'll update the comment.